### PR TITLE
Add support for numeric fields.

### PR DESCRIPTION
**Pull request by scolsen**
This PR adds support for numeric fields (currently limited to Doubles)

Screen Shot 2021-05-11 at 2 35 38 PM
Screen Shot 2021-05-11 at 2 35 51 PM
Screen Shot 2021-05-11 at 2 36 23 PM

Fixes issue #23

### CODE REVIEW

**Comment by scolsen** [comment]
@gino-m

File: gnd/src/main/java/com/google/android/gnd/model/observation/NumberResponse.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+/** A user provided response to a text {@link Field}. */
+public class NumberResponse implements Response {
+
+  private Number number;
+
+  public NumberResponse(Number number) {
+    this.number = number;
+  }
+
+  //TODO: Support other numeric types, such as int.
```
Comment (gino-m) [inline gnd/src/main/java/com/google/android/gnd/model/observation/NumberResponse.java:32]: Nit: Please file as FR instead

File: gnd/src/main/java/com/google/android/gnd/model/observation/NumberResponse.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+ * limitations under the License.
+ */
+
+package com.google.android.gnd.model.observation;
+
+import androidx.annotation.NonNull;
+import com.google.android.gnd.model.form.Field;
+import java8.util.Optional;
+
+/** A user provided response to a text {@link Field}. */
```
Comment (gino-m) [inline gnd/src/main/java/com/google/android/gnd/model/observation/NumberResponse.java:23]: text->number

File: gnd/src/main/java/com/google/android/gnd/model/observation/NumberResponse.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  }
+
+  @NonNull
+  @Override
+  public String toString() {
+    return number.toString();
+  }
+
+  public static Optional<Response> fromNumber(Number number) {
+    return number.toString().isEmpty()
```
Comment (gino-m) [inline gnd/src/main/java/com/google/android/gnd/model/observation/NumberResponse.java:75]: I think you can just do `return number.map(NumberPresponse::new)`

File: gnd/src/main/java/com/google/android/gnd/ui/editobservation/NumberFieldViewModel.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+public class NumberFieldViewModel extends AbstractFieldViewModel {
+
+  @Inject
+  NumberFieldViewModel(Application application) {
+    super(application);
+  }
+
+  public void updateResponse(String number) {
+    // TODO: Support specifying other numeric type restrictions, like INTEGER.
```
Comment (gino-m) [inline gnd/src/main/java/com/google/android/gnd/ui/editobservation/NumberFieldViewModel.java:31]: Same here re ticket.

**Review by gino-m — CHANGES_REQUESTED** [review]
Very cool, thanks for getting this done so fast! A few minor tweaks, otherwise LGTM.

File: gnd/src/main/java/com/google/android/gnd/model/observation/NumberResponse.java (hunk 3 above)
Comment (scolsen) [reply gnd/src/main/java/com/google/android/gnd/model/observation/NumberResponse.java:75]: Unfortunately, there's no `map` on `Number`

File: gnd/src/main/java/com/google/android/gnd/model/observation/NumberResponse.java (hunk 2 above)
Comment (scolsen) [reply gnd/src/main/java/com/google/android/gnd/model/observation/NumberResponse.java:23]: Good catch, thanks!

File: gnd/src/main/java/com/google/android/gnd/ui/editobservation/NumberFieldViewModel.java (hunk 4 above)
Comment (scolsen) [reply gnd/src/main/java/com/google/android/gnd/ui/editobservation/NumberFieldViewModel.java]: done

**Comment by scolsen** [comment]
@gino-m small bump!

**Review by gino-m — APPROVED** [review]
LGTM, thanks for the quick implementation! 🚀

**Pull request merged**